gsadjust/GSadjust.py

```diff
@@ -659,23 +680,24 @@ def workspace_open_getjson(self):
         :return:
         """
         fname, _ = QtWidgets.QFileDialog.getOpenFileName(None, 'Open File', self.path_data)
-        if fname:
-            if fname[-4:] != '.gsa':
-                self.msg = show_message('Saved workspaces should have a .gsa extension. ' +
-                                        'Please use "Open raw...data" to load a data file' +
-                                        ' or "Open workspace (.p format)" to open a workspace' +
-                                        ' with .p extension.', 'File load error')
-                return
+        if not fname or fname[-4:] != '.gsa':
```

Comment:
As a general rule it's better to test your failure states and exit, leaving the main path through your code as the "default".  We can combine the tests on this line `not fname or fname[-4:] != '.gsa'` as if `fname` is `None` the or short-circuits, and will not perform the second test.

If we get past this line we know that `fname` is non-False, and it does not end with `.gsa` which is what we want.